[Finishes #164829220 users can share links on social media

vino-jasuba opened the pull request:
#### Description
Currently, users have no way to share articles on other social media sites. This feature adds functionality to enable users to do this.

#### Type of change
- [ ] Bug fix (nonbreaking change which fixes an issue)
- [x] New feature (nonbreaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

#### How Has This Been Tested?
Please describe the tests that you ran to verify your changes
- [ ] End to end
- [ ] Integration

#### Checklist:
- [x] My code follows the style guide for this project
- [x] I have linted my code prior to submission
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

#### Pivotal Tracker
[164829220](https://www.pivotaltracker.com/story/show/164829220)

vino-jasuba committed:
164829220 feature(share): users can share articles

 - ensure users can share articles on social media and emails

[Delivers #164829220]

vino-jasuba committed:
164829220 feature(share): users can share articles

- ensure users can share articles on social media and emails

[Delivers #164829220]

ogiste commented on authors/apps/articles/serializers.py:
```diff
@@ -32,6 +34,28 @@ def to_representation(self, instance):
+            links['facebook'] = "https://www.facebook.com/sharer/sharer.php?u={}".format(article_link)
+            links['google_plus'] = "https://plus.google.com/share?url={}".format(article_link)
+            links['twitter'] = "https://twitter.com/home?status={}".format(article_link)
+            links['email'] = "mailto:?&subject={}&body={}".format(title, description, article_link)
```
What does the email link do?

vino-jasuba replied:
It's supposed to open the default mail client for the user so that they can share articles via email.

ogiste reviewed (commented):
Fix the builds and look into the email link.

ogiste commented on authors/apps/articles/serializers.py:
```diff
@@ -32,6 +34,28 @@ def to_representation(self, instance):
+            description = parse.quote(description)
+
+            links['facebook'] = "https://www.facebook.com/sharer/sharer.php?u={}".format(article_link)
+            links['google_plus'] = "https://plus.google.com/share?url={}".format(article_link)
```
Great work. Please remove the google plus link as google plus is officially shutting down personal and brand accounts.

vino-jasuba committed:
164829220 fix(share links): users can share articles

 - remove google plus from share_links
 - add article title as text for twitter share url

[Delivers #164829220]

bl4ck4ndbr0wn reviewed (approved):
Nice implementation.
Good work @vino-jasuba

oagutu commented on authors/apps/authentication/tests/test_mail.py:
```diff
@@ -41,8 +41,9 @@ def test_can_verify_mail(self):
-            r'http:\/\/[a-zA-Z0-9]+\/[a-z]+\/[a-z]+-[a-z]+\/[-0-9a-z]+\/[A-z]+[0-9]+\/',
+            r'http:\/\/[a-zA-Z0-9]+\/[a-z]+\/[a-z]+-[a-z]+\/[-0-9a-z]+\/\w+\/',
             mail.outbox[0].body)
+        print("first", mail.outbox[0].body)
```
Omit the print statements in this file

oagutu reviewed (commented):
@vino-jasuba This looks good, just make the suggested minor change
